src/coreclr/jit/morphblock.cpp

```diff
@@ -1272,7 +1272,12 @@ GenTree* MorphCopyBlockHelper::CopyFieldByField()
             dstFld = m_comp->gtNewLclvNode(dstFieldLclNum, m_comp->lvaGetDesc(dstFieldLclNum)->TypeGet());
 
             // If it had been labeled a "USEASG", assignments to the individual promoted fields are not.
-            dstFld->gtFlags |= m_dstLclNode->gtFlags & ~(GTF_NODE_MASK | GTF_VAR_USEASG);
+            dstFld->gtFlags |= m_dstLclNode->gtFlags & ~(GTF_NODE_MASK | GTF_VAR_USEASG | GTF_VAR_DEATH_MASK);
+
+            if (m_dstLclNode->IsLastUse(i))
+            {
+                dstFld->gtFlags |= GTF_VAR_DEATH;
```

Comment:
We can avoid creating IR for these stores at all (#80498). I'll handle that in a follow-up.